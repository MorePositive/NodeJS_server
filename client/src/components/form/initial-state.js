export const initialState = {
  owner: "",
  model: "",
  mark: "",
  year: ""
};
