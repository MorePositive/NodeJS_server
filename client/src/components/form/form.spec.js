import { shallow } from 'enzyme';
import React from 'react';

import Form from './form';
import { initialState } from './initial-state';

describe("Form component", () => {
  let component;
  let instance;
  const event = { preventDefault: () => {} };
  const mockCallBack = jest.fn();

  beforeEach(() => {
    component = shallow(<Form />);
    instance = component.instance();
  });

  it("should render Form component", () => {
    const wrapper = component.find("form");
    expect(wrapper.length).toEqual(1);
    expect(component).toMatchSnapshot();
  });
  
  it("should get initial state", () => {
    instance.getInitialState();
    expect(component.state()).toStrictEqual(initialState);
  });

  it("should handle input value", () => {
    expect(component.state().firstName).toBe("");
    instance.handleChange({ target: { name: "firstName", value: "test" } });
    expect(component.state().firstName).toBe("test");
  });

  describe("should validate email", () => {
    it("should return false if email field is empty", () => {
      instance.setState({ email: ""});
      const result = instance.validate();
      expect(result).toBe(false);
    });

    it("should return true if email field isn't empty", () => {
      instance.setState({ email: "test@test.com"});
      const result = instance.validate();
      expect(result).toBe(true);
    });
  });

  it("should not reload page after submiting form", () => {
    jest.spyOn(event, 'preventDefault')
    component.find("form").simulate("submit", event);
    expect(event.preventDefault).toBeCalled();
  });

  it("onCancel: should cancel the form field value", () => {
    instance.setState({ phone: 12455678, email: "test@test.com" });
    component.find(".cancel-btn").simulate("click", event);
    expect(component.state().phone).toBe("");
    expect(component.state().email).toBe("");
  });

  it("should call onClear method", () => {
    component = shallow(<Form clearData={mockCallBack} />);
    component.find(".clear-btn").simulate("click", event);
    expect(mockCallBack.mock.calls.length).toBe(1);
  });
});
