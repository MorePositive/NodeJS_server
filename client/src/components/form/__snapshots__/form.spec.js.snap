// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render Form component 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="form-wrapper"
    >
      <label
        className="label"
        htmlFor="firstName"
      >
        First name
      </label>
      <input
        className="form-input"
        name="firstName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-wrapper"
    >
      <label
        className="label"
        htmlFor="lastName"
      >
        Last name
      </label>
      <input
        className="form-input"
        name="lastName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-wrapper"
    >
      <label
        className="label"
        htmlFor="phone"
      >
        Phone number
      </label>
      <input
        className="form-input"
        name="phone"
        onChange={[Function]}
        type="tel"
        value=""
      />
    </div>
    <div
      className="form-wrapper"
    >
      <label
        className="label"
        htmlFor="email"
      >
        Email
        <span
          className="required"
        >
          *
        </span>
      </label>
      <input
        className="form-input"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
      <div
        className=""
      />
    </div>
    <div
      className="buttons-group"
    >
      <button
        className="btn add-btn"
        type="submit"
      >
        Add Item
      </button>
      <button
        className="btn cancel-btn"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="btn getdata-btn"
      >
        Get data
      </button>
      <button
        className="btn clear-btn"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
</form>
`;
